hugo/content/docs/dependency_injection.md

```diff
@@ -0,0 +1,114 @@
+        configureViewerOptions(context, {
+            needsClientLayout: false,
+            baseDiv: containerId
+        });
+    });
+    const container = new Container();
+    loadDefaultModules(container);
+    container.load(myModule, edgeIntersectionModule);
+    return container;
+};
+```
+The container is built from multiple modules. Through `loadDefaultModules()` all modules are loaded for default Sprotty functionalities. we can of course also load other optional modules like the `edgeIntersectionModule` for extra functionality.
+
+Most important is our own module where the core of the configuration happens. Here we can configure singleton scope classes like our [datasource](../datasources) or rebind default Sprotty components, for example the logger, to a custom implementation. We use Symbols for bindings instead of using classes directly. All Symbols can be found in the `TYPES` object.
+
+Through `configureModelElement` we can link our model to specific view components through the type property. Meaning if we have 
+```Typescript
+<SNode & TaskNode>{
+            type: 'task',
+            id: 'task01',
+            name: 'First Task',
+            isFinished: true,
+            ...
+}
+```
+in our model, Sprotty will try to convert this data structure to an instance of the actual `SNode` class and render it with the `TaskNodeView`.
+
+Lastly, we configure our viewer options. Here we configure all the DOM elements needed by Sprotty for example the base `div` inside of which our diagram is rendered or the hidden `div` used by the first render cycle for determining micro layout. Another thing configured here is layouting. Specifically, if layouting should be done on client-side, server-side or both. This also determines the protocol spoken by client and server. 
+
+## Features
+Model elements can further be configured through features. 
+```typescript
+configureModelElement(context, 'task', SNode, TaskNodeView, {
+    enable: [customFeature],
+    disable: [moveFeature]
+});
+```
+the `configureModelElement` method takes as an optional last parameter an object containing arrays for `enabled` and `disabled` features containing Symbols representing those features. Through this, we can disable default functionality like dragging or selecting nodes and add functionality, either custom or loaded through other non-default modules.
+
+## Creating custom components
+As described previously Sprotty uses inversifyJs for dependency injection. That means that when creating custom features, views, etc. we have to use it too.
+As an example let's look at Sprotty's `PolylineEdgeView`.
+
+```Typescript
+@injectable()
+export class PolylineEdgeView extends RoutableView {
+
+    @inject(EdgeRouterRegistry) edgeRouterRegistry: EdgeRouterRegistry;
+    ...
+}
+```
+The most important thing for our component to be available in Sprotty is annotating it with `@injectable()`. Otherwise the dependency injection won't work.
```

Comment:
```suggestion
The most important thing here for our component to be made available in Sprotty is to annotate it with `@injectable()`. Otherwise the dependency injection won't work.
```